Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp

```diff
@@ -3556,15 +3556,19 @@ void MediaPlayerPrivateGStreamer::triggerRepaint(GstSample* sample)
     {
         Locker sampleLocker { m_sampleMutex };
         shouldTriggerResize = !m_sample;
-        m_sample = sample;
+        m_sample = WTFMove(sample);
     }
 
     if (shouldTriggerResize) {
         GST_DEBUG_OBJECT(pipeline(), "First sample reached the sink, triggering video dimensions update");
-        GRefPtr<GstCaps> caps(gst_sample_get_caps(sample));
-        if (!caps) {
-            GST_ERROR_OBJECT(pipeline(), "Received sample without caps: %" GST_PTR_FORMAT, sample);
-            return;
+        GRefPtr<GstCaps> caps;
+        {
+            Locker sampleLocker { m_sampleMutex };
```

Comment:
I think it would be better to lock above and keep the lock until before `if (!m_canRenderingBeAccelerated) {`. Then there you can `unlockEarly()` . Then I think we would be good to go unless there is something I'm missing.

Reply:
I would avoid invasive changes here, unless you found a possible deadlock case. Is it the case?

Reply:
I didn't find a dead lock, but it is weird to see you're dumping a lock on something you're going to grab again microseconds later.